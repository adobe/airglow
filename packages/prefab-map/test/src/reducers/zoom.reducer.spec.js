/*
Copyright 2019 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

import { buttonZoomReducer, zoomReducer } from '../../../src/reducers/zoom.reducer';

describe('zoomReducer', () => {
  it('sets the zoom value', () => {
    expect(zoomReducer({}, { payload: { zoom: 7 } })).toMatchSnapshot();
  });
});

describe('buttonZoomReducer', () => {
  const state = { construct: { defaultZoom: 8 } };
  it('zooms of a default value', () => {
    expect(buttonZoomReducer({}, { payload: { size: -1 } })).toMatchSnapshot();
  });
  it('zooms off the provided default', () => {
    expect(buttonZoomReducer(state, { payload: { size: 1 } })).toMatchSnapshot();
  });
  it('clamps the zoom level', () => {
    expect(buttonZoomReducer(state, { payload: { size: 100 } })).toMatchSnapshot();
  });
});
