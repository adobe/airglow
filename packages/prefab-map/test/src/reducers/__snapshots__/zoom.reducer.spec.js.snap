// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttonZoomReducer clamps the zoom level 1`] = `
Object {
  "construct": Object {
    "defaultZoom": 8,
  },
  "store": Object {
    "zoom": 18,
  },
}
`;

exports[`buttonZoomReducer zooms of a default value 1`] = `
Object {
  "store": Object {
    "zoom": 3,
  },
}
`;

exports[`buttonZoomReducer zooms off the provided default 1`] = `
Object {
  "construct": Object {
    "defaultZoom": 8,
  },
  "store": Object {
    "zoom": 9,
  },
}
`;

exports[`zoomReducer sets the zoom value 1`] = `
Object {
  "store": Object {
    "zoom": 7,
  },
}
`;
